### PR TITLE
Force Windows SDK to be 10.0

### PR DESCRIPTION
**Pull request by jkotas**
This fixes the build failures on Windows Server 2012

### CODE REVIEW

**Comment by bartonjs** [comment]
Applying this did allow me to build the coreclr half of the project.  I'm not quite far enough along to know if that prevented it from executing on 2012R2.

**Review by jkoritzinsky — APPROVED** [review]

**Review by jaredpar — APPROVED** [review]

File: src/coreclr/build.cmd [hunk 1]
```diff
@@ -567,7 +562,7 @@ if %__BuildNative% EQU 1 (
 
     echo %__MsgPrefix%Regenerating the Visual Studio solution
 
-    set __ExtraCmakeArgs=!___SDKVersion! !___CrossBuildDefine! "-DCLR_CMAKE_PGO_INSTRUMENT=%__PgoInstrument%" "-DCLR_CMAKE_OPTDATA_PATH=%__PgoOptDataPath%" "-DCLR_CMAKE_PGO_OPTIMIZE=%__PgoOptimize%" "-DCLR_COMMON_DIR=%__RepoRootDir%/eng/common"
+    set __ExtraCmakeArgs="-DCMAKE_SYSTEM_VERSION=10.0" !___CrossBuildDefine! "-DCLR_CMAKE_PGO_INSTRUMENT=%__PgoInstrument%" "-DCLR_CMAKE_OPTDATA_PATH=%__PgoOptDataPath%" "-DCLR_CMAKE_PGO_OPTIMIZE=%__PgoOptimize%" "-DCLR_COMMON_DIR=%__RepoRootDir%/eng/common"
```
Comment (jaredpar) [inline src/coreclr/build.cmd]: @dagood does the official build path converge here? Or do we need to update the official build in a similar way?

Comment (jkotas) [reply src/coreclr/build.cmd]: The official build uses this same path.

**Review by danmoseley — APPROVED** [review]
thank you!

**Review by bartonjs — APPROVED** [review]
With this change I was able to build both coreclr and libraries and run tests with libraries on 2012R2.

Assuming that testing libraries (/t:BuildAndTest) uses the coreclr I built this didn't make the coreclr build Win10-only.

**Review by stephentoub — APPROVED** [review]

**Pull request merged**